src/tags/object/Paragraphs/HtxParagraphs.js

```diff
@@ -397,6 +413,41 @@ class HtxParagraphsView extends Component {
         return false;
       });
     });
+
+
+    if (isFF(FF_LSDV_E_278) && this.props.item.contextscroll && item.playingId >= 0 && this.lastPlayingId !== item.playingId && this.state.canScroll) {
+      const _start = this.props.item._value[item.playingId].start;
+      const _end = this.props.item._value[item.playingId].end;
+      const _phaseHeight = root.querySelectorAll('div')[item.playingId]?.offsetHeight || 0;
```

Comment:
is there anyway you can get a ref to `root.querySelectorAll('div')[item.playingId]` instead of querying it through the DOM? it would be more efficient that navigating the DOM to get this

Reply:
It's being created in another component, it would be possible but not in a correct way I guess.

Reply:
Querying the whole list each time to find the indexed id to pinpoint the active paragraph segment could be quite inefficient, especially depending on how long the list becomes and may also produce unwanted results if this list is ever virtualized or handled differently within React itself. If we were to instead pass an activeRef and set it within the child to point to the current active segment:

```tsx
// child paragraph
ref={isPlaying && activeRef}
```
that would work as well, then you can instead just check:

```tsx
// Paragraphs parent
activeRef.current?.offsetHeight || 0;
```